platform/linux-generic/odp_schedule_basic.c

```diff
@@ -2053,42 +2055,42 @@ static void schedule_print(void)
 	ODP_PRINT("  prefer ratio:      %u\n", sched->config.prefer_ratio);
 	ODP_PRINT("\n");
 
-	ODP_PRINT("  Number of active event queues:\n");
-	ODP_PRINT("              spread\n");
-	ODP_PRINT("            ");
+	pos  = 0;
+	pos += ODP_SNPRINT(&str[pos], size - pos, "  Number of active event queues:\n");
```

Comment:
I think the code needed in the users of buffered logging gets cluttered and fragile. Like this in the scheduler patch:

```
int pos;
const int size = 512;
char str[size]:
...
pos = 0;
pos += ODP_SNPRINT(&str[pos], size - pos, "  Number of active event queues:\n");
pos += ODP_SNPRINT(&str[pos], size - pos, "              spread\n");
pos += ODP_SNPRINT(&str[pos], size - pos, "            ");
...
ODP_PRINT("%s\n", str);
```

Wouldn't something like this be much better:
```
_odp_print_buffered("  Number of active event queues:\n");
_odp_print_buffered("              spread\n");
_odp_print_buffered("            ");
...
_odp_print_flush();
```

Or, if one wants to reduce the number of code changes needed, there could be a new function to enable buffered logging and then the existing ODP_PRINT() calls would change to buffered ones and only the flushing would be needed.

The implementation of the above schemes would have a thread-local buffer that the caller would not need to manage or much care about.

Reply:
Sure. However, I'd consider that as future development on top of this PR. This adds a safer way to use snprinf() (we have over 200 calls to snprintf(), many without return value check), and fixes an immediate issue of fragmented output from sched print.